src/m365/spo/commands/term/term-list.ts

```diff
@@ -108,36 +115,105 @@ class SpoTermListCommand extends SpoCommand {
-        headers: {
-          'X-RequestDigest': res.FormDigestValue
-        },
-        data: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="70" ObjectPathId="69" /><ObjectIdentityQuery Id="71" ObjectPathId="69" /><ObjectPath Id="73" ObjectPathId="72" /><ObjectIdentityQuery Id="74" ObjectPathId="72" /><ObjectPath Id="76" ObjectPathId="75" /><ObjectPath Id="78" ObjectPathId="77" /><ObjectIdentityQuery Id="79" ObjectPathId="77" /><ObjectPath Id="81" ObjectPathId="80" /><ObjectPath Id="83" ObjectPathId="82" /><ObjectIdentityQuery Id="84" ObjectPathId="82" /><ObjectPath Id="86" ObjectPathId="85" /><Query Id="87" ObjectPathId="85"><Query SelectAllProperties="false"><Properties /></Query><ChildItemQuery SelectAllProperties="true"><Properties><Property Name="Name" ScalarProperty="true" /><Property Name="Id" ScalarProperty="true" /></Properties></ChildItemQuery></Query></Actions><ObjectPaths><StaticMethod Id="69" Name="GetTaxonomySession" TypeId="{981cbc68-9edc-4f8d-872f-71146fcbb84f}" /><Method Id="72" ParentId="69" Name="GetDefaultSiteCollectionTermStore" /><Property Id="75" ParentId="72" Name="Groups" />${termGroupQuery}<Property Id="80" ParentId="77" Name="TermSets" />${termSetQuery}<Property Id="85" ParentId="82" Name="Terms" /></ObjectPaths></Request>`
-      };
-
-      const processQuery: string = await request.post(requestOptions);
-      const json: ClientSvcResponse = JSON.parse(processQuery);
-      const response: ClientSvcResponseContents = json[0];
-      if (response.ErrorInfo) {
-        throw response.ErrorInfo.ErrorMessage;
+      const result = await this.executeCsomCall(data, spoAdminUrl, res);
+      const terms: Term[] = [];
+      if (result._Child_Items_ && result._Child_Items_.length > 0) {
+        for (const term of result._Child_Items_) {
+          this.setTermDetails(term);
+          terms.push(term);
+          if (args.options.includeChildTerms && term.TermsCount > 0) {
+            await this.getChildTerms(spoAdminUrl, res, term, terms);
+          }
+        }
       }
 
-      const result: TermCollection = json[json.length - 1];
-      if (result._Child_Items_ && result._Child_Items_.length > 0) {
-        result._Child_Items_.forEach(t => {
-          t.CreatedDate = new Date(Number(t.CreatedDate.replace('/Date(', '').replace(')/', ''))).toISOString();
-          t.Id = t.Id.replace('/Guid(', '').replace(')/', '');
-          t.LastModifiedDate = new Date(Number(t.LastModifiedDate.replace('/Date(', '').replace(')/', ''))).toISOString();
-        });
-        logger.log(result._Child_Items_);
+      if (!args.options.output || args.options.output === 'json') {
+        logger.log(terms);
+      }
+      else {
+        // Converted to text friendly output
+        if (!args.options.includeChildTerms) {
```

Comment:
can we refactor to avoid nesting if conditions? If I am not mistaken we may rewrite this as
``` typescript
if (...) {
...
} else if (...) {
...
} else {
...
}
```

Reply:
Will refactor!